### PR TITLE
Proofreading

### PR DESCRIPTION
**Pull request by SomeoneSerge**
I'm going to read spec.md (reading just in spare time, which is little) and then maybe to try to reimplement Tox Protocol based on the specification (just for learning, you know).
As I advance in reading I'm going to fix typos, leave notes and ask questions -- I'll be pushing changes into this PR

Some questions are placed in the paper and marked with "TODO:" prefix

At the moment I'd propose adding goals and threat model part into the beginning of specification, so that it can serve as purpose and intuition for the rest of document and future reimplementations.
I also noted that threat model isn't clearly stated anywhere, which leads to some misunderstandings in discussions (e.g. this rather implicit tradeoff between anonymity and things like realtime AV)

Also I wanted to know if [kademlia](http://pdos.csail.mit.edu/~petar/papers/maymounkov-kademlia-lncs.pdf)'s least-recent-seen eviction policy (see diffs or referenced paper for rationale) was considered and if it was why was it rejected

Also, I disclaim, ugh, everything (see commitmsg for all these "I'm not ...")

### CODE REVIEW

**Comment by iphydf** [comment]
Thanks, can you check the formatting? Our travis build reformats the code in a canonical way. Perhaps you can adopt its formatting. If you feel that it's doing the wrong thing (i.e. it's producing semantically incorrect code), let me know.

File: spec.md [hunk 1]
```diff
@@ -305,17 +368,18 @@ connect directly to them via UDP.
...
+binary operator `+` and the identity element `0`.
 
-The DHT needs a [metric](https://en.wikipedia.org/wiki/Metric_(mathematics)) to
+The DHT uses a [metric](https://en.wikipedia.org/wiki/Metric_(mathematics)) to
 determine distance between two nodes. The Distance type is the co-domain of
 this metric. The metric currently used by the Tox DHT is the `XOR` of the
-nodes' public keys. The public keys are interpreted as Big Endian integers (see
-[Crypto Numbers](#key-1)).
+nodes' public keys: `distance(x, y) = x XOR y`.
+Public keys are interpreted as Big Endian integers (see[Crypto Numbers](#key-1)).
```
Comment (iphydf) [inline spec.md]: Add space after "see".

File: spec.md [hunk 2]
```diff
@@ -325,17 +389,31 @@ without computing the complete integral value. This works, because as soon as
 an ordering decision can be made in the most significant bits, further bits
 won't influence that decision.
 
-The XOR metric `d` satisfies the required conditions:
+The XOR is a valid metric, i.e. it satisfies the required conditions:
```
Comment (iphydf) [inline spec.md]: Remove "The".

**Review by iphydf — CHANGES_REQUESTED** [review]

File: spec.md [hunk 3]
```diff
@@ -375,14 +444,15 @@ set, the last (greatest) element is the furthest away.
 
 ### Bucket Index
 
-The bucket index can be computed using the following function:
+The index of bucket can be computed using the following function:
```
Comment (iphydf) [inline spec.md]: Why? Also, this should now be "The index of the bucket", but then this is exactly the same as "The bucket index", so why the change?

File: spec.md [hunk 4]
```diff
@@ -540,7 +615,7 @@ response has in their list of known nodes.
 
 ### Packed node format
 
-The DHT Send nodes uses the Packed Node Format.
+The DHT Send Nodes uses Packed Node Format.
```
Comment (iphydf) [inline spec.md]: Why did you remove "the"?

**Review by iphydf — CHANGES_REQUESTED** [review]

File: spec.md (hunk 3 above)
Comment (SomeoneSerge) [reply spec.md]: Er, that's right.
Rationale for the change was, as noted in the commit message, that while reading first time I'd been confused  if it was "[The bucket]'s index" or "The [bucket index] of something else contained in bucket"? (though it was possible to derive from context).
I admit, this can be just my misunderstanding of grammar/semantics, as English is not my native

File: spec.md (hunk 4 above)
Comment (SomeoneSerge) [reply spec.md]: Ugh, not sure; It was late night... Maybe I felt like it's kind of name so no need for?

File: spec.md (hunk 3 above)
Comment (SomeoneSerge) [reply spec.md]: Can we just add another "the" like you proposed?

File: spec.md (hunk 4 above)
Comment (SomeoneSerge) [reply spec.md]: I'll undo this one

**Review by iphydf — APPROVED** [review]

**Comment by iphydf** [comment]
LGTM, can you squash all commits into one, and give it some nice commit message?
